src/spawn.c

```diff
@@ -871,13 +893,24 @@ static gboolean spawn_write_cb(GIOChannel *channel, GIOCondition condition, gpoi
 }
 
 
+static gboolean spawn_read_cb(GIOChannel *channel, GIOCondition condition, gpointer data);
+
+static gboolean spawn_timeout_read_cb(gpointer data)
+{
+	SpawnChannelData *sc = (SpawnChannelData *) data;
+
+	/* The solution for continuous empty G_IO_IN-s is to generate them outselves. :) */
+	return spawn_read_cb(sc->channel, G_IO_IN, data);
```

Comment:
Shouldn't we attempt to become a normal callback again, instead of staying on the timeout source indefinitely?

Reply:
> It would have been sooo much easier if glib had source suspend/resume...

can't one simply ref the source, remove it from the context, add it back and unref it as a mean to pause it?